src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala

```diff
@@ -59,15 +63,20 @@ final case class PurposeVersionActivation(
       waitingForApprovalVersion <- purposeManagementService.waitForApprovalPurposeVersion(
         purpose.id,
         draftVersion.id,
-        StateChangeDetails(ChangedBy.CONSUMER)
+        StateChangeDetails(ChangedBy.CONSUMER, Some(dateTimeSupplier.get()))
       )
     } yield waitingForApprovalVersion
 
     def activate(version: PurposeVersion, changedBy: ChangedBy): Future[PurposeVersion] = {
+      val newSuspendedAt =
+        if (purpose.suspendedByConsumer.contains(true) && purpose.suspendedByProducer.contains(true))
```

Comment:
if we keep the timestamp optional, this change is required
```suggestion
        if (purpose.suspendedByConsumer.contains(true) || purpose.suspendedByProducer.contains(true))
```

Despite being correct having the business logic in the process, at the moment is based in the management.
Splitting the logic may be confusing, in my opinion it would be better for now to keep the logic in the management, until we migrate it all